src/libraries/System.Security.Cryptography.Pkcs/src/System/Security/Cryptography/Pkcs/SignedCms.cs

```diff
@@ -215,6 +201,34 @@ internal void Decode(ReadOnlyMemory<byte> encodedMessage)
 
             Version = _signedData.Version;
             _hasData = true;
+
+            static byte[] CopyContent(ReadOnlySpan<byte> encodedMessage)
+            {
+                unsafe
+                {
+                    fixed (byte* pin = encodedMessage)
```

Comment:
This PR introduced several new unsafe code paths just so that a `ROS<byte>` could be smuggled as a `ROM<byte>` because that's what `ContentInfoAsn.Decode` expects. If this is going to be a continuing pattern, I'd prefer instead if we did something like this, which would keep all of the unsafe code contained to a single method:

```cs
internal ref struct ValueContentInfoAsn
{
    public string ContentType;
    public ReadOnlySpan<byte> Content;

    internal static ValueContentInfoAsn Decode(ref AsnValueReader reader, ROS<byte> span)
    {
        // move the unsafe PointerMemoryManager logic here
        // call ContentInfoAsn.Decode with the custom MemoryManager
        return /* ... */;
    }
}
```

Reply:
The problem, of course, is that it's an infectious feature; and I'd need to write support for it in the code generator.  It would definitely allow for some simplified code calling (and reduce risk), and I have some ideas for how to do it without too much overhead.

So it'll definitely be on my mind as a cleanup effort / future impact.